src/items.cpp

```diff
@@ -1273,93 +1259,92 @@ bool dropItem(Item* item, int player, bool notifyMessage)
-		entity->flags[UPDATENEEDED] = true;
-		entity->x = players[player]->entity->x;
-		entity->y = players[player]->entity->y;
-		entity->sizex = 4;
-		entity->sizey = 4;
-		entity->yaw = players[player]->entity->yaw;
-		entity->vel_x = (1.5 + .025 * (rand() % 11)) * cos(players[player]->entity->yaw);
-		entity->vel_y = (1.5 + .025 * (rand() % 11)) * sin(players[player]->entity->yaw);
-		entity->vel_z = (-10 - rand() % 20) * .01;
-		entity->flags[PASSABLE] = true;
-		entity->behavior = &actItem;
-		entity->skill[10] = item->type;
-		entity->skill[11] = item->status;
-		entity->skill[12] = item->beatitude;
-		entity->skill[13] = qtyToDrop;
-		entity->skill[14] = item->appearance;
-		entity->skill[15] = item->identified;
-		entity->parent = players[player]->entity->getUID();
-		entity->itemOriginalOwner = entity->parent;
+	}
 
-		// play sound
-		playSoundEntity( players[player]->entity, 47 + rand() % 3, 64 );
+	Entity* entity = newEntity(-1, 1, map.entities, nullptr); //Item entity.
+	entity->flags[INVISIBLE] = true;
+	entity->flags[UPDATENEEDED] = true;
+	entity->x = players[player]->entity->x;
+	entity->y = players[player]->entity->y;
+	entity->sizex = 4;
+	entity->sizey = 4;
+	entity->yaw = players[player]->entity->yaw;
+	entity->vel_x = (1.5 + .025 * (rand() % 11)) * cos(players[player]->entity->yaw);
+	entity->vel_y = (1.5 + .025 * (rand() % 11)) * sin(players[player]->entity->yaw);
+	entity->vel_z = (-10 - rand() % 20) * .01;
+	entity->flags[PASSABLE] = true;
+	entity->behavior = &actItem;
+	entity->skill[10] = item->type;
+	entity->skill[11] = item->status;
+	entity->skill[12] = item->beatitude;
+	entity->skill[13] = qtyToDrop;
+	entity->skill[14] = item->appearance;
+	entity->skill[15] = item->identified;
+	entity->parent = players[player]->entity->getUID();
+	entity->itemOriginalOwner = entity->parent;
+
+	// play sound
+	playSoundEntity( players[player]->entity, 47 + rand() % 3, 64 );
+
+	// unequip the item
+	Item** slot = itemSlot(stats[player], item);
+	if ( slot != nullptr )
+	{
+		*slot = nullptr;
+	}
 
-		// unequip the item
-		Item** slot = itemSlot(stats[player], item);
-		if ( slot != NULL )
-		{
-			*slot = NULL;
-		}
-		if ( item->node != NULL )
+	if ( item->node != nullptr )
+	{
+		if ( item->node->list == &stats[0]->inventory )
 		{
-			if ( item->node->list == &stats[0]->inventory )
+			oldcount = item->count;
+			item->count = qtyToDrop;
+			if ( notifyMessage )
 			{
-				oldcount = item->count;
-				item->count = qtyToDrop;
-				if ( notifyMessage )
-				{
-					messagePlayer(player, language[1088], item->description());
-				}
-				item->count = oldcount - qtyToDrop;
-				if ( item->count <= 0 )
-				{
-					list_RemoveNode(item->node);
-					return true;
-				}
+				messagePlayer(player, language[1088], item->description());
 			}
-		}
-		else
-		{
-			item->count = item->count - qtyToDrop;
+			item->count = oldcount - qtyToDrop;
 			if ( item->count <= 0 )
 			{
-				free(item);
+				list_RemoveNode(item->node);
```

Comment:
This line shows how difficult PRs are sometimes, the PR is correct but github shows this changing a line earlier in the code.. would have been a breaking change. This is fine

Reply:
Very odd why it says something changed there.